File: src/main/java/de/twometer/amongus/gui/CallMeetingPage.java
package de.twometer.amongus.gui;

import de.twometer.amongus.core.AmongUs;
import de.twometer.amongus.net.NetMessage;
import de.twometer.neko.util.Log;

public class CallMeetingPage extends BasePage {

    public CallMeetingPage() {
        super("CallMeeting.html");
    }

    @Override
    public void onDomReady() {
        super.onDomReady();
        context.call("setCallsLeft", AmongUs.get().getSession().getMyself().emergencyMeetings);
    }

    public void callEmergency() {
        Log.i("Emergency Meeting!");
        AmongUs.get().getClient().sendMessage(new NetMessage.CallEmergency(NetMessage.EmergencyCause.Button));
        AmongUs.get().getSession().getMyself().emergencyMeetings--;
    }

    @Override
    protected boolean escapeGoesBack() {
        return true;
    }
}

File: src/main/kotlin/de/twometer/amogus/gui/CallMeetingPage.kt
package de.twometer.amogus.gui

import de.twometer.amogus.client.AmongUsClient
import de.twometer.amogus.net.CallMeeting

class CallMeetingPage : BasePage("CallMeeting.html") {

    override fun onLoaded() {
        call("setCallsLeft", AmongUsClient.session!!.myself.emergencyMeetings)
    }

    fun callEmergency() {
        AmongUsClient.send(CallMeeting(true))
        AmongUsClient.session!!.myself.emergencyMeetings--
    }

}